### DISCUSSION TITLE
Some Error in Rendering or so

### DISCUSSION THREAD

**Discussion by svn-josh** [TomSchimansky/CustomTkinter#1894] [General]
**so i got a small scrolable button frame wiht some entrys and a search bar:**
![image](https://github.com/TomSchimansky/CustomTkinter/assets/39739909/643fd041-bd8d-46d4-84f3-95701b455e91)

```python
# create archive frame & search bar
self.archiv_frame = customtkinter.CTkFrame(self, corner_radius=0, fg_color="transparent")
self.archiv_frame.rowconfigure(1, weight=1)
self.archiv_frame.columnconfigure(0, weight=1)
self.archiv_frame_scrollable_button_frame = ScrollableLabelButtonFrame(
master=self.archiv_frame, width=300, corner_radius=0, command=self.archiv_button_reprint_event)
self.archiv_frame_scrollable_button_frame.grid(row=1, column=0, columnspan=2, padx=20, pady=20, sticky="nsew")

self.archiv_frame_search_entry = customtkinter.CTkEntry(self.archiv_frame)
self.archiv_frame_search_entry.grid(row=0, column=0, padx=20, pady=[20, 0], sticky="ew")

self.search_button = customtkinter.CTkButton(self.archiv_frame, text="Suchen", image=self.search_image, command=self.search_archive_thread)
self.search_button.grid(row=0, column=1, padx=[0, 20], pady=[20, 0], sticky="e")

def search_archive_thread(self):
    t = threading.Thread(target=self.search_archive)
    t.start()

def search_archive(self):
    search_term = self.archiv_frame_search_entry.get()
    archive_list = self.ftp.get_archiv_list()
    if archive_list:
        self.archiv_frame_scrollable_button_frame.clear_items()
        for item in archive_list:
            if search_term.lower() in item.lower():
                self.archiv_frame_scrollable_button_frame.add_item(item)
 ```

When Clicking the Search Button once, everything is fine, but when clicking a second time it throws a error i think is related to some kind of Rendering

**Error:** https://pastebin.com/Ji0MjUBZ

**Comment by svn-josh** [comment]
Also my code again but this time in pastebin: https://pastebin.com/RbPPnZdB

**Comment by IfTrueEqualsEqualsTrue** [comment]
This is probably due to the use of Threading. Modifying widgets in thread differents from main thread is dangerous because of the non-thread-safe property of Tkinter.
Try in a first place to remove the thread !

**Reply by svn-josh** [reply]
the problem without using theading ist whenever the search button is pressed the whole app times out and gets no response from windows for like 10sek ...... hmm....

**Reply by ElectricCandlelight** [reply]
You can still use threads just don't do any tkinter operations in a different thread. You can still use event_generate in a different thread to act as a callback
